spyder/app/mainwindow.py

```diff
@@ -1434,9 +1434,30 @@ def post_visible_setup(self):
         self.menuBar().raise_()
         self.is_setting_up = False
 
+        # Handle DPI scale changes and show restart message
+        screen = self.window().windowHandle().screen()
+        screen.logicalDotsPerInchChanged.connect(self.show_DPI_change_message)
+
         # Notify that the setup of the mainwindow was finished
         self.sig_setup_finished.emit()
 
+    def show_DPI_change_message(self):
+        """Show message to restart Spyder since the DPI scale changed."""
+        screen = self.window().windowHandle().screen()
+        screen.logicalDotsPerInchChanged.disconnect(
+            self.show_DPI_change_message)
+        QMessageBox.warning(
+            self, _("Warning"),
+            _("A monitor scale change was detected. <br><br>"
+              "For a proper display Spyder will be restarted."),

```

Comment:
Please change this line to

    Spyder will be restarted to be properly displayed.